src/main/js/dynamic/loaded/certificates/CertificatesPage.js

```diff
@@ -31,125 +35,157 @@ class CertificatesPage extends Component {
         if (certificatesToDelete) {
-            certificatesToDelete.forEach(certificateId => {
-                this.props.deleteCertificate(certificateId);
+            certificatesToDelete.forEach((certificateId) => {
+                deleteCertificateAction(certificateId);
             });
         }
         this.setState({
             deleteCallback: callback
         });
     }
 
+    onEdit(selectedRow, callback) {
+        this.setState({
+            certificate: selectedRow
+        }, callback);
+    }
+
+    onCopy(selectedRow, callback) {
+        const copy = JSON.parse(JSON.stringify(selectedRow));
+        copy.id = null;
+        this.setState({
+            certificate: copy
+        }, callback);
+    }
+
     createModalFields() {
         const { certificate } = this.state;
         const { fieldErrors } = this.props;
         const aliasKey = 'alias';
         const certificateContentKey = 'certificateContent';
         return (
             <div>
-                <ReadOnlyField label="Last Updated" name="lastUpdated" readOnly="true" value={certificate['lastUpdated']} />
+                <ReadOnlyField
+                    id="lastUpdated"
+                    label="Last Updated"
+                    name="lastUpdated"
+                    readOnly="true"
+                    value={certificate.lastUpdated}
+                />
                 <TextInput
-                    name={aliasKey} label="Alias" description="The certificate alias name."
-                    required onChange={this.handleChange} value={certificate[aliasKey]}
+                    id={aliasKey}
+                    name={aliasKey}
+                    label="Alias"
+                    description="The certificate alias name."
+                    required
+                    onChange={this.handleChange}
+                    value={certificate[aliasKey]}
                     errorName={aliasKey}
-                    errorValue={fieldErrors[aliasKey]} />
+                    errorValue={fieldErrors[aliasKey]}
+                />
                 <TextArea
-                    name={certificateContentKey} label="Certificate Content" description="The certificate content text."
-                    required onChange={this.handleChange} value={certificate[certificateContentKey]}
+                    id={certificateContentKey}
+                    name={certificateContentKey}
+                    label="Certificate Content"
+                    description="The certificate content text."
+                    required
+                    onChange={this.handleChange}
+                    value={certificate[certificateContentKey]}
                     errorName={certificateContentKey}
-                    errorValue={fieldErrors[certificateContentKey]} />
+                    errorValue={fieldErrors[certificateContentKey]}
+                />
             </div>
         );
     }
 
-    onEdit(selectedRow, callback) {
+    handleChange(e) {
```

Comment:
We might have this exact function in another utility class. May be useful importing and using it

Reply:
I think we may have talked about creating a utility for this, but I'm not finding a utility for the change event.

Reply:
I think you're right. Just took a quick look and didn't see any. Perhaps it's a nice future change.